v3/undefined/exif_C4A5_print_image_matching.go

```diff
@@ -0,0 +1,81 @@
+package exifundefined
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+
+	log "github.com/dsoprea/go-logging"
+
+	exifcommon "github.com/dsoprea/go-exif/v3/common"
+)
+
+var PrintImageMatchingHeader = []byte{0x50, 0x72, 0x69, 0x6e, 0x74, 0x49, 0x4d, 0x00}
+
+type TagC4A5PrintImageMatching struct {
+	Version string
+	Value   []byte
+}
+
+func (TagC4A5PrintImageMatching) EncoderName() string {
+	return "CodecC4A5PrintImageMatching"
+}
+
+func (ev TagC4A5PrintImageMatching) String() string {
+	return fmt.Sprintf("TagC4A5PrintImageMatching<VERSION=(%s) BYTES=(%v)>", ev.Version, ev.Value)
+}
+
+type CodecC4A5PrintImageMatching struct{}
+
+func (CodecC4A5PrintImageMatching) Encode(value interface{}, byteOrder binary.ByteOrder) (encoded []byte, unitCount uint32, err error) {
+	defer func() {
+		if state := recover(); state != nil {
+			err = log.Wrap(state.(error))
+		}
+	}()
+
+	pim, ok := value.(TagC4A5PrintImageMatching)
+	if !ok {
+		log.Panicf("can only encode a TagC4A5PrintImageMatching")
+	}
+
+	return pim.Value, uint32(len(pim.Value)), nil
+}
+
+func (CodecC4A5PrintImageMatching) Decode(valueContext *exifcommon.ValueContext) (value EncodeableValue, err error) {
+	// PIM structure explanation: https://www.ozhiker.com/electronics/pjmt/jpeg_info/pim.html Looks like
+	// there is no sufficient open documentation about this tag:
+	// https://github.com/Exiv2/exiv2/issues/1419#issuecomment-739723214
+	// For that reason we are just preserving its raw bytes
+	defer func() {
+		if state := recover(); state != nil {
+			err = log.Wrap(state.(error))
+		}
+	}()
+
+	ev := TagC4A5PrintImageMatching{}
+
+	valueContext.SetUndefinedValueType(exifcommon.TypeByte)
+	rawBytes, err := valueContext.ReadBytes()
+	ev.Value = rawBytes
```

Comment:
The documentation describes how to find an entry count and the list of entries. We can parse things better than just yielding an opaque byte slice, yes?

Reply:
I could not find any decent documentation on PrintIM. It is not specified in the exif 2.2 spec. The only thing I found was  the link I posted above and it does not agree with my images. My images have a 106 byte PrintIM. According to the spec this would mean 

Header: 8
Version: 5
ExtraNull: 1
EntryCount: 2
Entries: 6*EntryCount

This works out if there are 15 entries. However here is the actual PrintIM bytes of one of my photos:

```
00000000: 5072 696e 7449 4d00 3033 3030 0000 0300  PrintIM.0300....
00000010: 0200 0100 0000 0300 2200 0000 0101 0000  ........".......
00000020: 0000 0911 0000 1027 0000 0b0f 0000 1027  .......'.......'
00000030: 0000 9705 0000 1027 0000 b008 0000 1027  .......'.......'
00000040: 0000 011c 0000 1027 0000 5e02 0000 1027  .......'..^....'
00000050: 0000 8b00 0000 1027 0000 cb03 0000 1027  .......'.......'
00000060: 0000 e51b 0000 1027 0000 0a              .......'...
```

Bytes 15-16 are `[03 00]` which yields 3 entries (order is Little Endian). So it does not work out.

Let's say we ignore the count and treat the remaining bytes as 6 byte entries. Here is what we get:

```
0200 0100 0000
0300 2200 0000
0101 0000 0000
0911 0000 1027
0000 0b0f 0000
1027 0000 9705
0000 1027 0000
```
and so on.

If the first 2 bytes of each entry would designate its tag id we have 2 tag ids of `[00 00]` which does not make much sense as we have a duplication as well as a meaningless tag id.

That's why I gave up and decided I would rather keep the tag opaque rather than parse it incorrectly.